magefiles/docker.go

```diff
@@ -108,20 +108,40 @@ func runDocker(
 		containerConfig.User = fmt.Sprintf("%d:%d", os.Getuid(), os.Getgid())
 	}
 
-	resp, err := docker.ContainerCreate(ctx, &containerConfig, &container.HostConfig{
-		AutoRemove: true,
-		Mounts: []mount.Mount{
-			{
-				Type:   mount.TypeBind,
-				Source: cwd,
-				Target: "/opt",
-			},
-			{
-				Type:   mount.TypeBind,
-				Source: "/lib/modules",
-				Target: "/lib/modules",
-			},
+	mounts := []mount.Mount{
+		{
+			Type:   mount.TypeBind,
+			Source: cwd,
+			Target: "/opt",
+		},
+		{
+			Type:   mount.TypeBind,
+			Source: "/lib/modules",
+			Target: "/lib/modules",
 		},
+	}
+
+	if env["MOUNT_HOST_GOMODCACHE"] == "1" {
+		// `GOMODCACHE` is not accessible from go code directly during build, therefore it
+		// has to be acquired in a separate process.
+		out, err := exec.Command("go", "env", "GOMODCACHE").CombinedOutput()
+		if err != nil {
+			fmt.Println("Error while executing 'go env GOMODCACHE':", err)
```

Comment:
Does it make sense to continue here in case of an error? Similarly in the line 133. If we have issues with getting `GOMODCACHE` then why still add the mount?